src/main/java/io/nats/client/Dispatcher.java

```diff
@@ -33,6 +33,14 @@
  */
 public interface Dispatcher extends Consumer {
 
+    /**
+     * Start the dispatcher with a given id.
+     * Use post-construction to start to the dispatcher,
+     * which should not be started on construction.
+     * @param id the assigned id of the dispatcher
+     */
+    void start(String id);

```

Comment:
Needed to add this so impl could override it. The connection sometimes creates a dispatcher and does not start it - it's created lazy - and wanted to ensure the slim edge that 2 threads try to create at the same time.